File: Data Structures & Algorithms/asteroid-collision/submission-1.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<Integer>();

        for(int ast: asteroids){
            while(!stack.isEmpty() && ast < 0 && stack.peek() > 0){
                int diff = ast + stack.peek();
                if(diff > 0){
                    ast = 0;
                }
                else if(diff < 0){
                    stack.pop();
                }
                else{
                    ast = 0;
                    stack.pop();
                }
            }
            if(ast != 0){
                stack.push(ast);
            }
        }

        return stack.stream().mapToInt(i->i).toArray();
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket_stack = []
        for char in s:
            if char == '}':
                if bracket_stack and bracket_stack[-1] == '{':
                    bracket_stack.pop()
                else:
                    return False
            elif char == ')':
                if bracket_stack and bracket_stack[-1] == '(':
                    bracket_stack.pop()
                else:
                    return False
            elif char == ']':
                if bracket_stack and bracket_stack[-1] == '[':
                    bracket_stack.pop()
                else:
                    return False
            else:
                bracket_stack.append(char)
        
        if len(bracket_stack) == 0:
            return True
        return False